tools/crd-bumper/crd-bumper.py

```diff
@@ -0,0 +1,292 @@
+  createapis.edit_new_api_files()
+  createapis.remove_previous_storage_version()
+  createapis.set_storage_version()
+  createapis.add_conversion_schemebuilder()
+
+  MAKE.fmt()
+  createapis.commit_copy_api_content(git, stage)
+  return True
+
+
+def mv_webhooks(git, stage, project, args):
+  """
+  Move the webhooks from the previous hub to the new hub.
+  """
+
+  webhooks = MvWebhooks(args.dryrun, project, args.prev_ver, args.new_ver, CGEN.preferred_api_alias(), CGEN.module())
+  webhooks.edit_go_files()
+  webhooks.edit_manifests()
+  webhooks.mv_project_webhooks()
+
+  MAKE.fmt()
+  webhooks.commit(git, stage)
+  return True
+
+
+def conversion_webhooks(git, stage, project, args):
+  """
+  Add a conversion webhook to anything in the new hub that needs it.
+  """
+
+  webhooks = ConversionWebhooks(args.dryrun, project, args.prev_ver, args.new_ver, CGEN.preferred_api_alias(), CGEN.module())
+  webhooks.create()
+  CGEN.fix_kubebuilder_import_alias()
+  webhooks.hub()
+  webhooks.enable_in_crd()
+  webhooks.add_fuzz_tests()
+
+  MAKE.fmt()
+  webhooks.commit(git, stage)
+  return True
+
+
+def conversion_gen(git, stage, project, args):
+  """
+  Create the spoke conversion routines and tests.
+  """
+  cgen = CGEN
+  cgen.mk_doc_go()
+  cgen.mk_spoke()
+  cgen.mk_spoke_fuzz_test()
+  cgen.mk_conversion_webhook_suite_test()
+
+  MAKE.fmt()
+  cgen.commit(git, stage)
+  return True
+
+
+def bump_controllers(git, stage, project, args):
+  """
+  Bump controllers to new hub version
+  """
+
+  controllers = Controllers(args.dryrun, project, args.prev_ver, args.new_ver, CGEN.preferred_api_alias(), CGEN.module())
+  controllers.run()
+  controllers.edit_util_conversion_test()
+
+  MAKE.fmt()
+  controllers.commit_bump_controllers(git, stage)
+  return True
+
+
+def bump_apis(git, stage, project, args):
+  """
+  Bump earlier spoke APIs to new hub version.
+  """
+
+  controllers = Controllers(args.dryrun, project, args.prev_ver, args.new_ver, CGEN.preferred_api_alias(), CGEN.module(), "api")
+  bumped = controllers.bump_earlier_spokes()
+
+  MAKE.fmt()
+  if bumped:
+    controllers.commit_bump_apis(git, stage)
+  else:
+    git.verify_clean()
+  return bumped
+
+
+def auto_gens(git, stage, project, args):
+  """
+  Make auto-generated files.
+  """
+
+  makecmd = MAKE
+  makecmd.update_spoke_list()
+  makecmd.manifests()
+  makecmd.generate()
+  makecmd.generate_go_conversions()
+  makecmd.fmt()
+
+  makecmd.commit(git, stage)
+  return True
+
+
+operation_order = [
+  ["create-apis",         create_apis,         None],
+  ["copy-api-content",    copy_api_content,    None],
+  ["mv-webhooks",         mv_webhooks,         None],
+  ["conversion-webhooks", conversion_webhooks, None],
+  ["conversion-gen",      conversion_gen,      None],
+  ["bump-controllers",    bump_controllers,    None],
+  ["bump-apis",           bump_apis,           None],
+  ["auto-gens",           auto_gens,
+    {"alternate-prev": "bump-controllers"},
+  ],
+]
+
+
```

Comment:
I think you should wrap all of this in a main function and put it at the top. I find this a bit hard to read with the function definitions sprinkled throughout the main code. It's a bit too "go-like".

I'd get a better idea of what's going on with a main at the top, then I can dive deeper into what the individual functions are doing.

Reply:
It has a main() now.

Go-like...hmmmpf.